go/arrow/flight/flightsql/server.go

```diff
@@ -511,8 +511,36 @@ func (BaseServer) BeginSavepoint(context.Context, ActionBeginSavepointRequest) (
 	return nil, status.Error(codes.Unimplemented, "BeginSavepoint not implemented")
 }
 
-func (BaseServer) CancelQuery(context.Context, ActionCancelQueryRequest) (CancelResult, error) {
-	return CancelResultUnspecified, status.Error(codes.Unimplemented, "CancelQuery not implemented")
+func (b *BaseServer) CancelQuery(context context.Context, request ActionCancelQueryRequest) (CancelResult, error) {
+	result, err := b.CancelFlightInfo(context, request.GetInfo())
+	if err != nil {
+		return CancelResultUnspecified, err
+	}
```

Comment:
So what we could do is check whether or not the server they passed has a `CancelQuery` method, and if it does then we dispatch to that. Otherwise, default to calling `CancelFlightInfo`

Reply:
You mean that we keep backward compatibility for users who are using `CancelQuery`, right?
If so, could you push a code for it to this branch? I don't know much about `reflect` package in Go...